tests/autodiff/reverse-single-iter-loop.slang

```diff
@@ -0,0 +1,99 @@
+//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -shaderobj -output-using-type
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj -output-using-type
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0 0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<float> outputBuffer;
+
+typedef DifferentialPair<float> dpfloat;
+typedef float.Differential dfloat;
+
+[BackwardDifferentiable]
+float test_simple_single_iter_loop(float y)
+{
+    float x = 0;
+
+    for (int i = 0; i < 1; i++)
+    {
+        if (y > 0.6)
+        {
+            x = y * 10.0f;
+            break;
+        }
+        else
+        {
+            x = y * 6.0f;
+            break;
+        }
+    }
+    
+    return x;
+}
+
+[BackwardDifferentiable]
+float test_nested_if_else_single_iter_loop(float y)
+{
+    float x = 0;
+
+    for (int i = 0; i < 1; i++)
+    {
+        if (y > 0.6)
+        {
+            if (y > 0.8)
+            {
+                x = y * 10.0f;
+                break;
+            }
+            else
+            {
+                x = y * 4.0f;
+                break;
+            }
+        }
```

Comment:
We need to also test for statements that appear after the converge point of a loop, like here. This is where we should be inserting guarding `if`s.